shared/menubar/files.desktop.js

```diff
@@ -46,14 +43,55 @@ const FileUpdate = (props: FileUpdateProps) => (
   </Kb.Box2>
 )
 
-const FileUpdates = ({updates, moreUpdateCount}: FileUpdatesProps) => (
+type FileUpdatesState = {
+  isShowingAll: boolean,
+}
+
+const FileUpdatesHoc = (ComposedComponent: React.ComponentType<any>) =>
+  class extends React.PureComponent<FileUpdatesProps, FileUpdatesState> {
+    state = {
+      isShowingAll: false,
+    }
+    render() {
+      return <ComposedComponent {...this.props} onShowAll={() => this.setState({isShowingAll: !this.state.isShowingAll})} isShowingAll={this.state.isShowingAll} />
+    }
+  }
+
+type FileUpdatesHocProps = {|
+  onShowAll: () => void,
+  isShowingAll: boolean,
+|}
+
+type ShowAllProps = FileUpdatesHocProps & {|
+  numUpdates: number,
+|}
+
+const FileUpdatesShowAll = (props: ShowAllProps) => (
+  <Kb.Box2 direction="horizontal" fullWidth={true} centerChildren={false}>
+    <Kb.ClickableBox onClick={props.onShowAll} className="toggleButtonClass" style={styles.toggleButton}>
+      <Kb.Text type="BodySmallSemibold" style={styles.buttonText}>
+        {props.isShowingAll ? 'Collapse' : `+ ${(props.numUpdates - defaultNumFileOptionsShown).toString()} more`}
+      </Kb.Text>
+    </Kb.ClickableBox>
+  </Kb.Box2>
+)
+
+const defaultNumFileOptionsShown = 3
+
+const FileUpdates = (props: (FileUpdatesProps & FileUpdatesHocProps)) => (
   <Kb.Box2 direction="vertical" fullWidth={true}>
-    {updates.map(u => (
-      <FileUpdate key={u.name} {...u} />
+    {props.updates.slice(0, props.isShowingAll ? props.updates.length : defaultNumFileOptionsShown).map(u => (
+      <FileUpdate key={u.name} {...u} tlfType={props.tlfType} />
     ))}
+    {props.updates.length > defaultNumFileOptionsShown && (
+      // $FlowIssue ¯\_(ツ)_/¯
+      <FileUpdatesShowAll onShowAll={props.onShowAll} isShowingAll={props.isShowingAll} numUpdates={props.updates.length} />
+    )}
   </Kb.Box2>
 )
 
+const ComposedFileUpdates = compose(FileUpdatesHoc)(FileUpdates)
```

Comment:
This could just be `const ComposedFileUpdate = FileUpdatesHoc(FileUpdates)`. Also, it seems `onShowAll` and `isShowingAll` is only used in `FileUpdatesShowAll`, so if you choose to use HOC, it seems the HOC can wrap around `FileUpdatesShowAll` to avoid plumbing these through `FileUpdates`.

Reply:
Nope, `isShowingAll` is also used in `FileUpdates`: https://github.com/keybase/client/pull/13667/files#diff-41bacb263b756a2ef2365b2ef779dd93R83